### PR TITLE
Refactor as fix for #7

### LINKED ISSUE #7: code refactor basing on current structure
```
 java % pwd
/Users/yuanyi/OpenSource/java-gm/src/main/java
 java % ls
Interact.java   SM3Util.java    SM4Util.java
```
refactor needed for this structure.

### PR DESCRIPTION
**Pull request by SamYuan1990**
refactor
add test for SM2
add SM2 sign, verify
add SM2 encrypt, decrypt
close #7

### CODE REVIEW

**Comment by SamYuan1990** [comment]
@davidkhala , please help with PR.

**Comment by SamYuan1990** [comment]
btw, I am not familiar with crypto lib itself, so hope any one can review and suggest with the impl part.

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,5 @@ java-gm.iml
 .vscode
 build
 priv.pem
-pub.pem
\ No newline at end of file
+pub.pem
+todo
```
Comment (davidkhala) [inline .gitignore:8]: 这个todo是什么特殊文件？

Comment (SamYuan1990) [reply .gitignore]: updated

**Comment by SamYuan1990** [comment]
@davidkhala , any updates on this pr?

**Comment by davidatheals** [comment]
> @davidkhala , any updates on this pr?

can you merge into single commit?

**Comment by SamYuan1990** [comment]
> > @davidkhala , any updates on this pr?
> 
> can you merge into single commit?

just updated.

File: src/main/java/SM2Util.java [hunk 2]
```diff
@@ -23,45 +27,94 @@
...
- * @author 吴仙杰
- * @see <a href="https://www.pixelstech.net/article/1464167276-Generating-CSR-using-Java">Generating CSR using Java</a>
- * @see <a href="http://senthadev.com/generating-csr-using-java-and-bouncycastle-api.html">
- * Generating CSR using Java and BouncyCastle API</a>
- * @see <a href="https://github.com/Trisia/alg-sm2-demo">Trisia/alg-sm2-demo</a>
- */
-public class Interact {
-
-    private Interact() {
+ * @author SamYuan
```
Comment (davidatheals) [inline src/main/java/SM2Util.java:30]: 请勿删除原作者著作信息

File: src/main/java/SM2Util.java [hunk 3]
```diff
@@ -76,7 +129,7 @@ public static String generateCsr(boolean isRsaNotEcc) throws NoSuchAlgorithmExce
 
         // 使用私钥和 SHA256WithRSA/SM3withSM2 算法创建签名者对象
         ContentSigner signer = new JcaContentSignerBuilder(isRsaNotEcc ? "SHA256WithRSA" : "SM3withSM2")
-                .setProvider(BC)
+                //.setProvider(BC)
```
Comment (davidkhala) [inline src/main/java/SM2Util.java:132]: 需要删除的代码尽量直接删除而不是注释，如果在特定情况下这段代码需要uncomment，请单独阐述

File: src/main/java/SM2Util.java (hunk 2 above)
Comment (SamYuan1990) [reply src/main/java/SM2Util.java:30]: ok

File: src/main/java/SM2Util.java (hunk 3 above)
Comment (SamYuan1990) [reply src/main/java/SM2Util.java:132]: got

**Review by davidkhala — APPROVED** [review]
LGTM

**Pull request merged**